near-sdk-macros/src/core_impl/code_generator/item_impl_info.rs

```diff
@@ -810,4 +810,63 @@ mod tests {
         );
         assert_eq!(expected.to_string(), actual.to_string());
     }
+
+    #[test]
+    fn return_result_init() {
+        let impl_type: Type = syn::parse_str("Hello").unwrap();
+        let mut method: ImplItemMethod = parse_quote! {
+            #[init]
+            #[return_result]
+            pub fn new() -> Result<Self, &'static str> { }
+        };
+        let method_info = ImplItemMethodInfo::new(&mut method, impl_type).unwrap();
+        let actual = method_info.method_wrapper();
+        let expected = quote!(
+            #[cfg(target_arch = "wasm32")]
+            #[no_mangle]
+            pub extern "C" fn new() {
+                near_sdk::env::setup_panic_hook();
+                if near_sdk::env::attached_deposit() != 0 {
+                    near_sdk::env::panic_str("Method new doesn't accept deposit");
+                }
+                if near_sdk::env::state_exists() {
+                    near_sdk::env::panic_str("The contract has already been initialized");
+                }
+                let result = Hello::new();
+                match result {
+                    Ok(contract) => near_sdk::env::state_write(&contract),
+                    Err(err) => near_sdk::FunctionError::panic(&err)
+                }
+            }
+        );
+        assert_eq!(expected.to_string(), actual.to_string());
+    }
+
+    #[test]
+    fn return_result_init_ignore_state() {
+        let impl_type: Type = syn::parse_str("Hello").unwrap();
+        let mut method: ImplItemMethod = parse_quote! {
+            #[init(ignore_state)]
+            #[return_result]
+            pub fn new() -> Result<Self, &'static str> { }
```

Comment:
Yeah, so this annotation is a bit misleading when used with init because nothing is actually returned. I wonder if there is a better name for this annotation.

Reply:
Yeah, not the best name for sure. How about `#[handle_result_error]` or even `#[handle_result]`?

Reply:
> Yeah, not the best name for sure. How about `#[handle_result_error]` or even `#[handle_result]`?

I personally like the latter since it's more concise (result_error is a bit redundant), but up to you!